src/rpc/blockchain.cpp

```diff
@@ -1062,6 +1062,8 @@ static UniValue gettxoutsetinfo(const JSONRPCRequest& request)
 
     CCoinsStats stats;
     ::ChainstateActive().ForceFlushStateToDisk();
+
+    LOCK(::cs_main);
```

Comment:
In commit "Cover UTXO set access with lock annotations" (ceca2c92b79a8a2c5ae37d03e99de80ec55895c3)

Commit message is a bit misleading imo, because this is adding a new lock, not just new annotations.

But more importantly, GetUTXOStats is acquiring and releasing cs_main internally, so this change seems like it is complicating locking and also locking too much. Maybe a change like the following would make sense?

```diff
-    LOCK(::cs_main);
-    if (GetUTXOStats(&::ChainstateActive().CoinsDB(), stats)) {
+    CCoinsView* coins_view = WITH_LOCK(cs_main, return &ChainstateActive().CoinsDB());
+    if (GetUTXOStats(coins_view, stats)) {
```

Reply:
Ah that's a good point - I forgot that `GetUTXOStats` only briefly acquires cs_main to get the height for a certain blockhash; it (counterintuitively) doesn't need to hold cs_main to iterate through the coins db because leveldb's implicit use of snapshots with cursors. 

I think the `WITH_LOCK` usage above obeys the letter of the law but not the spirit since technically the coinsdb could be modified in between that line and the GetUTXOStats call, but in this context I don't think it matters. Thanks for the suggestion!